apps/api/src/app/inbox/usecases/snooze-notification/snooze-notification.usecase.ts

```diff
@@ -0,0 +1,242 @@
+        });
+
+      return snoozedNotification;
+    } catch (error) {
+      throw new InternalServerErrorException(`Failed to snooze notification: ${error.message}`);
+    }
+  }
+
+  public async enqueueJob(job: JobEntity, delay: number) {
+    this.logger.info({ jobId: job._id, delay }, 'Adding snooze job to Standard Queue', LOG_CONTEXT);
+
+    const jobData = {
+      _environmentId: job._environmentId,
+      _id: job._id,
+      _organizationId: job._organizationId,
+      _userId: job._userId,
+    };
+
+    await this.standardQueueService.add({
+      name: job._id,
+      data: jobData,
+      groupId: job._organizationId,
+      options: { delay, attempts: this.RETRY_ATTEMPTS, backoff: { type: 'exponential', delay: 5000 } },
+    });
+  }
+
+  private async isSnoozeEnabled(command: SnoozeNotificationCommand) {
+    const isSnoozeEnabled = await this.featureFlagsService.getFlag({
+      key: FeatureFlagsKeysEnum.IS_SNOOZE_ENABLED,
+      defaultValue: false,
+      organization: { _id: command.organizationId } as OrganizationEntity,
+      environment: { _id: command.environmentId } as EnvironmentEntity,
+      user: { _id: command.subscriberId } as UserEntity,
+    });
+
+    if (!isSnoozeEnabled) {
+      throw new NotImplementedException();
+    }
+
+    // TODO: add per environment feature on/off on integration settings
+  }
+
+  private calculateDelayInMs(snoozeUntil: Date): number {
+    return snoozeUntil.getTime() - new Date().getTime();
+  }
+
+  private async validateDelayDuration(command: SnoozeNotificationCommand, delay: number) {
+    const tierLimit = await this.getTierLimit(command);
+
+    if (tierLimit === -1) {
+      throw new HttpException(
+        {
+          message: 'Feature Not Available',
+          reason: 'Snooze functionality is not available on your current plan. Please upgrade to access this feature.',
+        },
+        HttpStatus.PAYMENT_REQUIRED
+      );
+    }
+
+    if (delay > tierLimit) {
+      throw new HttpException(
+        {
+          message: 'Snooze Duration Limit Exceeded',
+          reason:
+            'The snooze duration you selected exceeds your current plan limit. ' +
+            'Please upgrade your plan for extended snooze durations.',
+        },
+        HttpStatus.PAYMENT_REQUIRED
+      );
+    }
+  }
+
+  private async getTierLimit(command: SnoozeNotificationCommand) {
+    const organization = await this.organizationRepository.findOne({
+      _id: command.organizationId,
+    });
+
+    const systemLimitMs = await this.featureFlagsService.getFlag({
+      key: FeatureFlagsKeysEnum.MAX_DEFER_DURATION_IN_MS_NUMBER,
+      defaultValue: SYSTEM_LIMITS.DEFER_DURATION_MS,
+      environment: { _id: command.environmentId },
+      organization: { _id: command.organizationId },
+    });
+
+    const isSpecialLimit = systemLimitMs !== SYSTEM_LIMITS.DEFER_DURATION_MS;
+    if (isSpecialLimit) {
```

Comment:
im wondering if we should add a comment here, similar to what we did in ResourceValidatorService, because `Special` isn’t self-explanatory.

```suggestion
    // If the system limit is not the default, we need to use it as the absolute limit for special cases instead of the tier limit
    if (isSpecialLimit) {
```